test/scalability/runner/recorder/recorder.go

```diff
@@ -0,0 +1,558 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recorder
+
+import (
+	"context"
+	"encoding/csv"
+	"os"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/yaml"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/test/scalability/runner/generator"
+)
+
+type event interface {
```

Comment:
The metrics are not precise enough

Reply:
I think they will be precise enough for the use case of analyzing the scalability runs. Note that they are used for  scalability tests projects using ClusterLoader2. We could increase the scraping rate, but I think starting with 1s should be enough. The default for scraping is 15s, so this is anyway more frequent.

Metrics are routinely used for monitoring of prod, so we also want additional incentive to add introduce good metrics into Kueue, rather than crafting custom code. The metrics will benefit production environments.

Reply:
> I think they will be precise enough

Please explain.

Reply:
I just don't think drilling below 1s will give us much. The main focus for the metrics is to provide a manual tool to investigate what is going on during a run. We scrape metrics in kubernetes scalability tests with 15s resolution, and from what I heard this is pretty much enough.

Reply:
https://github.com/kubernetes-sigs/kueue/pull/1931#issuecomment-2049414703